src/bijectors/planar_layer.jl

```diff
@@ -23,74 +23,110 @@ function Base.:(==)(b1::PlanarLayer, b2::PlanarLayer)
     return b1.w == b2.w && b1.u == b2.u && b1.b == b2.b
 end
 
-function get_u_hat(u, w)
-    # To preserve invertibility
-    x = w' * u
-    return u .+ (planar_flow_m(x) - x) .* w ./ sum(abs2, w)   # from A.1
-end
-
 function PlanarLayer(dims::Int, wrapper=identity)
     w = wrapper(randn(dims))
     u = wrapper(randn(dims))
     b = wrapper(randn(1))
     return PlanarLayer(w, u, b)
 end
 
-planar_flow_m(x) = -1 + softplus(x)   # for planar flow from A.1
-ψ(z, w, b) = (1 .- tanh.(w' * z .+ b).^2) .* w    # for planar flow from eq(11)
+"""
+    get_u_hat(u::AbstractVector{<:Real}, w::AbstractVector{<:Real})
 
-# An internal version of transform that returns intermediate variables
-function _transform(flow::PlanarLayer, z::AbstractVecOrMat)
-    return _planar_transform(flow.u, flow.w, first(flow.b), z)
-end
-function _planar_transform(u, w, b, z)
-    u_hat = get_u_hat(u, w)
-    transformed = z .+ u_hat .* tanh.(w' * z .+ b) # from eq(10)
-    return (transformed = transformed, u_hat = u_hat)
-end
+Return a tuple of vector ``û`` that guarantees invertibility of the planar layer, and
+scalar ``wᵀ û``.
 
-(b::PlanarLayer)(z) = _transform(b, z).transformed
+# Mathematical background
 
-function forward(flow::PlanarLayer, z::AbstractVecOrMat)
-    transformed, u_hat = _transform(flow, z)
-    # Compute log_det_jacobian
-    psi = ψ(z, flow.w, first(flow.b)) .+ zero(eltype(u_hat))
-    if psi isa AbstractVector
-        T = eltype(psi)
-    else
-        T = typeof(vec(psi))
-    end
-    log_det_jacobian::T = log.(abs.(1 .+ psi' * u_hat)) # from eq(12)
-    return (rv = transformed, logabsdetjac = log_det_jacobian)
+According to appendix A.1, vector ``û`` defined by
+```math
+û(w, u) = u + (\\log(1 + \\exp{(wᵀu)}) - 1 - wᵀu) \\frac{w}{\\|w\\|²}
+```
+guarantees that the planar layer ``f(z) = z + û tanh(wᵀz + b)`` is invertible.
```

Comment:
```suggestion
guarantees that the planar layer ``f(z) = z + û tanh(wᵀz + b)`` is invertible even if `w, u ∈ ℝᵈ` and `b ∈ ℝ`.
```
It might be a bit redundant, it's just to emphasize that we're not assuming the inputs to be constrained in any way.

Reply:
```suggestion
guarantees that the planar layer ``f(z) = z + û tanh(wᵀz + b)`` is invertible for all ``w, u ∈ ℝᵈ`` and ``b ∈ ℝ``.
```